### PR TITLE
upgrade paramiko version to fix vulnerability

### PR DESCRIPTION
**Pull request by lesscodingmorehappiness**

### CODE REVIEW

File: microsoft/runbook/tiers/tier.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ extension:
   - ../../testsuites
 variable:
   - name: tier
-    value: 0
+    value: 1
```
Comment (squirrelsc) [inline microsoft/runbook/tiers/tier.yml]: Revert this change.

File: pyproject.toml [hunk 2]
```diff
@@ -12,10 +12,12 @@ classifiers = [
     "Topic :: Software Development :: Testing",
 ]
 dependencies = [
+    "temppathlib >= 1.2.0",
```
Comment (squirrelsc) [inline pyproject.toml]: Don't use `>=`. Check if both temppathlib and icontract should be in the list or not.

File: lisa/util/shell.py [hunk 3]
```diff
@@ -31,6 +51,638 @@
...
+        confirm: bool = True,
+    ):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.put(localpath, remotepath, callback, confirm))
+
+    def get(self, remotepath: str, localpath: str, callback: Optional[Callable[[int, int], Any]] = None):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.get(remotepath, localpath, callback))
+
+class SshShellWithSFTP(icontract.DBC):
```
Comment (squirrelsc) [inline lisa/util/shell.py:204]: What this is used to fix? Can we just include the core parts?

Comment (lesscodingmorehappiness) [reply lisa/util/shell.py:204]: yes I'm trying to only include the functions being used in lisa, I don't want to break current lisa functions or introduce unused contents.

BTW please ignore this draft for now I'm trying to fix some linting alerts

**Comment by squirrelsc** [comment]
Fixed by upgrading spurplus.
